src/search/pdbs/pattern_database.cc

```diff
@@ -258,8 +282,75 @@ void PatternDatabase::create_pdb(
             if (alternative_cost < distances[predecessor]) {
                 distances[predecessor] = alternative_cost;
                 pq.push(alternative_cost, predecessor);
+                if (compute_plan) {
+                    generating_op_ids[predecessor] = op_id;
+                    if (op.get_cost() == 0) {
+                        number_of_zero_cost_ops[predecessor] = number_of_zero_cost_ops[state_index] + 1;
+                    } else {
+                        number_of_zero_cost_ops[predecessor] = 0;
```

Comment:
Looks like they are. We might consider not doing that. This would make computing the PDB cheaper, but the found abstract plans would no longer be guaranteed to be strongly optimal, so it's a tradeoff. It's also worth considering disabling this computation in domains without zero-cost operators, since most of them don't have any.